### PR TITLE
[Fixes #10024] Wrong default style when creating layer

### LINKED ISSUE #10024: Wrong default style when creating layer
Sometimes when a layer is created, it's not referencing a brand new style, but has the default geoserver style (either `point`, `polygon`, etc).
Anyway, when GeoNode creates the wms calls, the style name is prefixed with `geonode:`
http://localhost/geoserver/ows?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng&TRANSPARENT=true&LAYERS=geonode%3Amalga_bb8b47bc672d71cd83776e209c0e2413&STYLES=geonode%3Apoint&SRS=EPSG%3A3857&CRS=EPSG%3A3857&TILED=true&_v_=1663259726834&access_token=yfFDxJAZQbw1ICp4iH3pfQs0DS6mxP&WIDTH=512&HEIGHT=512&BBOX=1252344.271424327%2C5322463.153553393%2C1565430.339280409%2C5635549.2214094745

and geoserver refuses it with this exception:

```xml
<ServiceExceptionReport>
   <ServiceException code="StyleNotDefined"> No such style: geonode:point </ServiceException>
</ServiceExceptionReport>
```

This problem seems to be a race condition, so it's quite complicated to replicate with a vanilla GeoNode.

A setup where this issue is always replicable is using the geonode-import app; when upload a geopackage file, more layers are created at once, so the race condition will happen even with a gpkg with only 2 layers.

## Steps to Reproduce the Problem
  1. setup a system with [geonode-importer](https://github.com/geosolutions-it/geonode-importer)
  2. upload a gpkg with at least 2 layers
  3. --> one of the layers will not be rendered because of the style problem

## Specifications

  - GeoNode version: 4.0.x
  - Installation method: docker

### PR DESCRIPTION
**Pull request by etj**
References: #10024

Solves the problem of bad styles assigned to layer.

- Main fix is by setting proper lock ids in celery tasks.
- Added lot of debug logging, also useful for post-mortem investigations
- Split some nested commands and improved robustness

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

### CODE REVIEW

**Comment by etj** [comment]
@afabiani can you also pls check for the duplication of the deletion of automatically created styles in GeoServer 
here
https://github.com/geosolutions-it/geonode/blob/d4ed55b890a4ea4b990cbeae5e2b48b9a5f544bb/geonode/geoserver/helpers.py#L1190-L1192
and here
https://github.com/geosolutions-it/geonode/blob/d4ed55b890a4ea4b990cbeae5e2b48b9a5f544bb/geonode/geoserver/helpers.py#L1216

**Comment by giohappy** [comment]
@afabiani it would also be great if you could add some comments to the steps, to help understanding what we're doing in each of them. It's very hard to follow the logic of style, default styles, etc. management

**Comment by giohappy** [comment]
From my side all looks good @etj. I was able to reproduce the issue before the fix. Now it never happens.

**Review by giohappy — APPROVED** [review]

**Review by afabiani — APPROVED** [review]
